### DISCUSSION TITLE
Circular Structure Error When passing complex objects with circular reference to another server component

### DISCUSSION THREAD

**Discussion by webplantmedia** [vercel/next.js#72225] [Help]
### Summary

I can't seem to pass complex objects to other server components or functions without getting a circular structure error. Specifically, I'm using html-react-parser and manipulating certain elements to render custom jsx. It worked fine and without error in nextjs 14. Can anyone confirm for me if this is how nextjs 15 will handle complex objects with circular reference? It breaks a major dependency I have and I need to figure out if I need to move in another direction with my libraries.

Here is an example of the error:

https://github.com/vercel/next.js/issues/72189
https://github.com/remarkablemark/html-react-parser/issues/1589

### Example

https://github.com/vercel/next.js/issues/72189

**Comment by icyJoseph** [comment]
This looks like a problem within React though.

I have a hunch that somehow, https://github.com/facebook/react/pull/31105 introduced it, but I can be totally off... nevertheless, when running your project on stackblitz, I got this stack:

```console
  2139 |       methodName = [methodName, stackTrace, owner, env];
  2140 |       methodName.push.apply(methodName, args);
> 2141 |       args = stringify(methodName, function (parentPropertyName, value) {
       |              ^
  2142 |         try {
  2143 |           return renderConsoleValue(
  2144 |             request,
```

Which makes me think, the error is here: https://github.com/facebook/react/blob/7c8e5e7ab8bb63de911637892392c5efd8ce1d0f/packages/react-server/src/ReactFlightServer.js - in any of the two instances of `renderConsoleValue` - give or that that Next.js uses compiled React dependencies, https://github.com/vercel/next.js/tree/canary/packages/next/src/compiled/react-server-dom-webpack


Note that, `next build && next start` do work, so I think this is a DEV mode issue.

Stackblitz fork: https://stackblitz.com/edit/github-12hutx?file=app%2Fpage.tsx
